src/CHIPBackend.cc

```diff
@@ -1086,14 +1086,12 @@ void chipstar::Context::syncQueues(chipstar::Queue *TargetQueue) {
     }
     SyncQueuesEvent = TargetQueue->enqueueBarrierImpl(EventsToWaitOn);
     SyncQueuesEvent->Msg = "barrierSyncQueue";
-    TargetQueue->updateLastEvent(SyncQueuesEvent);
```

Comment:
Why not consider the barrier's event the last one here? Is this safe to omit?

Reply:
@pjaaskel updated the description. Basically, all enqueue operations must execute `executeCommandList` which internally updates the last event.